modules/core/src/arithm.cpp

```diff
@@ -61,7 +61,7 @@ void convertAndUnrollScalar( const Mat& sc, int buftype, uchar* scbuf, size_t bl
 {
     int scn = (int)sc.total(), cn = CV_MAT_CN(buftype);
     size_t esz = CV_ELEM_SIZE(buftype);
-    getConvertFunc(sc.depth(), buftype)(sc.ptr(), 1, 0, 1, scbuf, 1, Size(std::min(cn, scn), 1), 0);
+    getConvertFunc(sc.depth(), CV_MAT_DEPTH(buftype))(sc.ptr(), 1, 0, 1, scbuf, 1, Size(std::min(cn, scn), 1), 0);
```

Comment:
since getConvertFunc() takes 2 ElemType's, let's insert 2 `CV_MAT_DEPTH()` calls into that function instead of using `CV_MAT_DEPTH()` each time we call it.